### ISSUE TITLE
False positive: Hoist statements out of for/while loops

### ISSUE THREAD

**Issue by kalanzun** [sourcery-ai/sourcery#53]
### Issue description or question

```python
import asyncio

class MyAwaitable:
    def __await__(self):
        print("__await__")
        yield

async def main():
    x = MyAwaitable()
    for _ in range(3):
        await x

asyncio.run(main())
```

This code prints "__await__" three times. Sourcery suggests to hoist statements out of for/while loops for `await x`:

```python
async def main():
    x = MyAwaitable()
    await x
```

This code prints "__await__" only once.

Side note: Sourcery does not suggest to hoist the statement if the operand of await is a call:

```python
import asyncio

class MyAwaitable:
    def __await__(self):
        print("__await__")
        yield

    def __call__(self):
        return self

async def main():
    x = MyAwaitable()
    for _ in range(3):
        await x()

asyncio.run(main())
```

This code prints "__await__" three times. asyncio (the module) seems to return awaitables from calls like `await asyncio.sleep(1)` most of the time if not always.

### Sourcery Version

Plan: Sourcery open Source. Plugin version: 0.3.7. Sourcery Version: 0.7.0.

### Code editor or IDE name and version

Pycharm v2020.1.1 (Professional Edition)

### OS name and version

Arch Linux

**Comment by brendanator** [comment]
Thanks for the report. Our analysis wasn't treating awaits specially. This has been fixed and will be in the next release

**Comment by Hellebore** [comment]
Fixed with vscode 0.2.5 and pycharm 0.4.0 releases.
